internal/manifests/collector/annotations.go

```diff
@@ -27,8 +27,8 @@ func Annotations(instance v1beta1.OpenTelemetryCollector) (map[string]string, er
 	// new map every time, so that we don't touch the instance's annotations
 	annotations := map[string]string{}
 
-	// Enable Prometheus annotations by default if DisablePrometheusAnnotations is nil or true
-	if !instance.Spec.Observability.Metrics.DisablePrometheusAnnotations {
+	// Enable Prometheus annotations by default if DisablePrometheusAnnotations|DisableAutomaticPrometheusAnnotations is nil or true
+	if !instance.Spec.Observability.Metrics.DisablePrometheusAnnotations || !instance.Spec.Observability.Metrics.DisableAutomaticPrometheusAnnotations {
```

Comment:
IMO we should only use the new line and just remove references to this other than the warning. Otherwise if the first annotation is unset this would always be true, nullifying the new flag